### PR TITLE
sql: remove defaultsettings role option

### PR DESCRIPTION
**Pull request by RichardJCai**
Should be a non-issue to remove as it was never actually
written to disk anywhere and only used for logging.

Release note: None

### CODE REVIEW

File: pkg/sql/roleoption/role_option.go [hunk 1]
```diff
@@ -60,7 +60,6 @@ const (
 	NOCANCELQUERY
 	MODIFYCLUSTERSETTING
 	NOMODIFYCLUSTERSETTING
-	DEFAULTSETTINGS
```
Comment (rafiss) [inline pkg/sql/roleoption/role_option.go:63]: just to confirm, we don't rely on the order of enums for this right?

File: pkg/sql/roleoption/role_option.go [hunk 2]
```diff
@@ -172,7 +170,7 @@ func (rol List) GetSQLStmts(op string) (map[string]func() (bool, string, error),
 		// outside of this set stmt.
 		// DEFAULTSETTINGS is stored in system.database_role_settings.
```
Comment (rafiss) [inline pkg/sql/roleoption/role_option.go:171]: could remove this comment

File: pkg/util/log/eventpb/role_events.proto [hunk 3]
```diff
@@ -54,6 +54,7 @@ message AlterRole {
   string role_name = 3 [(gogoproto.jsontag) = ",omitempty"];
   // The options set on the user/role.
   repeated string options = 4 [(gogoproto.jsontag) = ",omitempty", (gogoproto.moretags) = "redact:\"nonsensitive\""];
+  repeated string set_info = 5 [(gogoproto.jsontag) = ",omitempty", (gogoproto.moretags) = "redact:\"nonsensitive\""];
```
Comment (rafiss) [inline pkg/util/log/eventpb/role_events.proto:57]: this could use a comment -- is it specific for default settings?

**Review by rafiss — APPROVED** [review]
lgtm, feel free to address nits later if there's a rush.

also this would change the logging between 21.2 and 22.1, but i think that's fine?

**Comment by RichardJCai** [comment]
> lgtm, feel free to address nits later if there's a rush.
> 
> also this would change the logging between 21.2 and 22.1, but i think that's fine?

Actually we should either backport this to 21.2 or not backport this at all.

I don't see any major benefit to backporting this so perhaps we should just leave it?

**Comment by RichardJCai** [comment]
Thanks!
bors r=rafiss

**Comment by rafiss** [comment]
Yeah, let's not backport

**Pull request merged**